### PR TITLE
Migrate to pyproject.toml

### PR DESCRIPTION
**Pull request by SvenMarcus**
Moves from the older requirements files to the newer `pyproject.toml` project metadata format

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 authorized_keys
 __pycache__/
 .python-version
+.pdm.toml
```
Comment (markusweigelt) [inline .gitignore:5]:
```suggestion
.pdm.toml
.pdm.lock

```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+[project.optional-dependencies]
+dev = [
+    "beautifulsoup4>=4.11.1",
+    "mypy>=1.1.1",
+    "pytest>=7.2.2",
+    "pytest-asyncio>=0.21.0",
+    "testcontainers>=3.7.1",
+    "black>=23.1.0",
+    "types-requests>=2.28.11.15",
+    "types-beautifulsoup4>=4.12.0.0",
```
Comment (markusweigelt) [inline pyproject.toml:32]:
```suggestion
    "types-beautifulsoup4>=4.12.0.0",
    "types-Pygments>=2.14.0.6",
    "types-ujson>=5.7.0.1",
```

Comment (markusweigelt) [inline pyproject.toml:32]:
```suggestion
    "types-beautifulsoup4>=4.12.0.0",
    "types-Pygments>=2.14.0.6",
    "types-ujson>=5.7.0.1",
```

**Review by markusweigelt — CHANGES_REQUESTED** [review]
Great work! Here are a few suggestions for changes.

File: pyproject.toml (hunk 2 above)
Comment (SvenMarcus) [reply pyproject.toml:32]: Neither `pygments` nor `ujson` are installed in my virtual environment, so those dependencies should not be necessary.

Comment (markusweigelt) [reply pyproject.toml:32]: Ok then I check this with the CI pipeline whether the error appears still appears here.

**Review by markusweigelt — APPROVED** [review]

**Pull request merged**